Improve RPC documents

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40309 Add a warning to mention that async_execution does not work with autograd profiler
* #40305 Minor RPC doc improvements
* #40300 Fix RRef to_here() docs
* #40299 Fix RPC API doc links
* #40298 Improve docs for init_rpc
* **#40296 Improve RPC documents**
* #40291 Let torch.futures.wait_all re-throw errors

1. Added a link to parameter server tutorial
2. Explained current states for TorchScript support

Differential Revision: [D22142647](https://our.internmc.facebook.com/intern/diff/D22142647)

mrshenli committed:
Improve RPC documents

1. Added a link to parameter server tutorial
2. Explained current states for TorchScript support

[ghstack-poisoned]

rohan-varma commented on docs/source/rpc.rst:
```diff
@@ -97,7 +97,11 @@ applications can always explicitly move the input tensors to CPU on the caller
+  v1.5.0, ``torch.distributed.rpc`` supports calling TorchScript functions as
+  RPC target functions, and this will help improve parallelism on the callee
+  side as executing TorchScript functions does not require GIL. However, support
+  for calling RPC APIs within a TorchScript function is very limited.
```
I would say we should even leave out mentioning support for calling RPC APIs within a TS function, because of the following reasons:

1) We only added rpc_async which was mostly to support an internal use case, and so it was likely designed as such, but we may need additional special attention for OSS
2) I don't believe we have any docs/tutorials mentioning which APIs are supported/unsupported
3) Support for things such as profiler for RPC called within JIT does not work, and there may be other features that don't work either. For example, are we able to call an async UDF via rpc_async inside a torchscript function? 
4) It hasn't yet been through a design/feature review which new features normally go through

Is it okay for now for it to be unannounced, and if users do discover it they can play around with it, but we harden and officially target 1.7 to release this?

mrshenli replied:
Good point, let me remove the last sentence.

rohan-varma reviewed (approved):
LGTM although please consider the inline comment!

mrshenli committed:
Update on "Improve RPC documents"


1. Added a link to parameter server tutorial
2. Explained current states for TorchScript support

Differential Revision: [D22142647](https://our.internmc.facebook.com/intern/diff/D22142647)

[ghstack-poisoned]
